Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/FileListItem.java

```diff
@@ -29,7 +30,18 @@ public FileListItem(File file)
 		}
 		else
 		{
-			String fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			String fileExtension = null;
+
+			// If the file has no extension, this would throw an exception.
+			try
+			{
+				fileExtension = mPath.substring(mPath.lastIndexOf('.'));
+			}
+			catch (Exception ignored)
+			{
+				// We don't care about files without extensions anyway.
+				mType = TYPE_OTHER;
+			}
```

Comment:
I had just copied over the functionality from the old code, in `GameListFragment.fill()`, but this is a better way of doing it.

No point in changing it elsewhere though; the other places where this pattern is used will all be removed eventually.